.circleci/config.yml

```diff
@@ -343,7 +343,7 @@ jobs:
       - run: touch logs/build-status
       - run: 
           name: Skip build if possible
-          command: ruby scripts/should-build.rb || circleci step halt
+          command: ruby scripts/should_build.rb || circleci step halt
```

Comment:
Yes, but this is not just a cli tool, it's also a library file.

If we were instead to call it `scripts/should-build` with no file extension (which we _can_ do because it's `+x` and includes the `#!` line it needs) then RuboCop would not complain. I want to do that, if that sounds good to you.

Reply:
I would prefer that. 

Actually, if you wanted (probably in another PR), we could split this into a library file and an executable script – `should_build.rb` and `should-build`, and then `should-build` can require and invoke the function from `should_build.rb`.

Reply:
That's a better way to do it anyhow, because then we can mock `sh` and such functions and test our actual behavior!

I'll rename the script as the last commit in this PR, then merge when green.

Reply:
(speaking of merge-when-green… #3242)